src/main/groovy/io/codearte/gradle/nexus/infra/SimplifiedHttpJsonRestClient.groovy

```diff
@@ -1,16 +1,19 @@
 package io.codearte.gradle.nexus.infra
 
+import groovy.json.JsonOutput
+import groovy.json.JsonSlurper
 import groovy.transform.CompileStatic
 import groovy.util.logging.Slf4j
-import groovyx.net.http.ContentType
-import groovyx.net.http.HttpResponseDecorator
-import groovyx.net.http.HttpResponseException
-import groovyx.net.http.RESTClient
+import okhttp3.Credentials
+import okhttp3.Interceptor
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.RequestBody
+import okhttp3.Response
+import org.jetbrains.annotations.NotNull
 
 /**
  * Specialized REST client to communicate with Nexus.
- *
- * Note: The same as RESTClient from HTTP Builder this class is not thread-safe.
```

Comment:
I assume - the way how you reuse it - that backed by OkHttp this class is thread safe. You could mention it explicitly in JavaDoc.

Reply:
Sure, added the `@ThreadSafe` annotation